File: Services/Models/RoundEnd.cs
using OhMyWord.Core.Models;

namespace OhMyWord.Services.Models;

public class RoundEnd
{
    private readonly Round round;

    public Guid RoundId => round.Id;
    public RoundEndReason EndReason => round.EndReason;
    public DateTime NextRoundStart { get; }
    public string Word => round.Word.Value;

    internal RoundEnd(Round round, DateTime nextRoundStart)
    {
        this.round = round;
        NextRoundStart = nextRoundStart;
    }
}

File: Client/src/app/models/round-end.ts
import { RoundEndResponse } from './responses/round-end.response';

export class RoundEnd {
  roundId: string;
  nextRoundStart: Date;
  word: string;
  
  constructor(init?: Partial<RoundEndResponse>) {
    this.roundId = init?.roundId ?? '';
    this.nextRoundStart = init?.nextRoundStart ? new Date(init.nextRoundStart) : new Date();
    this.word = init?.word ?? '';
  }
}
